django/contrib/postgres/search.py

```diff
@@ -33,6 +35,24 @@ def db_type(self, connection):
         return 'tsquery'
 
 
+class SearchConfig(Expression):
+    def __init__(self, config):
+        super().__init__(output_field=None)
+        if not (config and hasattr(config, 'resolve_expression')):
+            config = Value(config)
+        self.config = config
+
+    def resolve_expression(self, query=None, allow_joins=True, reuse=None, summarize=False, for_save=False):
+        resolved = super().resolve_expression(query, allow_joins, reuse, summarize, for_save)
+        resolved.config = self.config.resolve_expression(query, allow_joins, reuse, summarize, for_save)
+        return resolved
```

Comment:
The usual pattern is to implement `get_source_expressions` and `set_source_expressions`.

```python
def get_source_expressions(self):
    return [self.config]

def set_source_expressions(self, expressions):
    self.config, = expressions

Reply:
Ah, makes sense!